### PR TITLE
Modal and Comparison Logic

### PR DESCRIPTION
**Pull request by Lordnickim**
Add Modal package and implemented in RelatedAction and App
Add Comparison Logic and Click handle implementation
First pass on CSS styling with inline

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -95,12 +110,14 @@ const App = () => {
         getOneProduct={getOneProduct}
         getStyles={getStyles}
         setCurrentProduct={setCurrentProduct}
+        currentProduct = {currentProduct}
```
Comment (raphaelspies) [inline client/src/components/App.jsx:105]: remove spaces on both sides of "=" symbol

File: client/src/components/RelatedAction.jsx [hunk 2]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
+height: 500px;
+display: absolute;
+align-items: center;
+justify-content: center;
+background-color: white;
 
+opacity: ${(props) => props.opacity};
+transition: all 0.3s ease-in-out;
+`
+
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx:25]: Remove unnecessary whitespace

File: client/src/components/RelatedAction.jsx [hunk 3]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
 
-export default function RelatedAction () {
-  const [isRelated, setIsRelated] = useState(true)
+  const beforeClose = () => {
+    return new Promise((resolve) => {
+      setOpacity(0);
+      setTimeout(resolve, 300);
+    });
+  }
+
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx:55]: remove unnecessary whitespace

File: client/src/components/RelatedAction.jsx [hunk 4]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
+
+  const comparisionData = () => {
+    const charArray = new Set();
+    let compareData = [];
+    console.log('ProductA Features:', product.features)
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: Remove unnecessary console logs when ready

File: client/src/components/RelatedAction.jsx [hunk 5]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
+    console.log('ProductB Features:', overviewProduct.features)
+    product.features.forEach(entry => charArray.add(entry.feature));
+    overviewProduct.features.forEach(entry => charArray.add(entry.feature));
+    console.log('CombinedFeatures:', charArray)
+
+    for(var i =0; charArray.size > i; i++) {
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: should be `let i=0` (remove spaces, change var => let)

File: client/src/components/RelatedAction.jsx [hunk 6]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
+        data.valueB = overviewProduct.features.find(entry => entry.feature === feature).value;
+        compareData.push(data);
+      } else if (productA.some(entry => entry.feature === feature)) {
+        console.log('Product A match')
+        data.feature = feature;
+        data.valueA = product.features.find(entry => entry.feature === feature).value;
+        data.valueB = null;
+        compareData.push(data);
+      } else {
+        console.log('Prodcut B match')
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: remove console.log when ready

Comment (raphaelspies) [reply client/src/components/RelatedAction.jsx]: Also, typo: prodcut instead of product

File: client/src/components/RelatedAction.jsx [hunk 7]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
+  }
 
   return (
-    <StyleImg src={isRelated? EmptyStar : CrossCircle}/>
+    <div>
+      <StyleImg onClick ={(e) => clickHandle(e)} src={isRelated? EmptyStar : CrossCircle}/>
+      <StyledModal
+        isOpen= {isOpen}
+        afterOpen={ afterOpen}
+        beforeClose = {beforeClose}
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: Remove spaces around =

File: client/src/components/RelatedAction.jsx [hunk 8]
```diff
@@ -10,12 +11,121 @@ const StyleImg = styled.img`
...
+            <th style= {{textAlign: "center"}}></th>
+            <th style= {{padding: "5px 8px 15px", textAlign: "right"}}>{overviewProduct.name}</th>
+          </tr>
+          </thead>
+          <tbody>
+          {compareInfo.map((entry, index) => (
+            <tr key={index}>
+              <td style= {{padding: "5px",textAlign: "left"}}>{entry.valueA === true ? "√" : entry.valueA}</td>
+              <td style= {{padding: "5px",textAlign: "center"}}>{entry.feature}</td>
+              <td style= {{padding: "5px",textAlign: "right"}}>{entry.valueB}</td>
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: Remove spaces around = sign

File: client/src/components/RelatedContainer.jsx [hunk 9]
```diff
@@ -97,7 +99,7 @@ export default function RelatedContainer ({id, getOneProduct, getStyles, setCurr
     <div style={{width: '50%'}}>
     <Carousel responsive= {responsive} centerMode= {true} >
       {productsInfo.map((entry, index) => (
-        <Card key={index} product={entry} setCurrentProduct={setCurrentProduct}/>
+        <Card key={index} product={entry} setCurrentProduct={setCurrentProduct} overviewProduct ={currentProduct}/>
```
Comment (raphaelspies) [inline client/src/components/RelatedContainer.jsx]: Remove space before = sign

**Review by raphaelspies — CHANGES_REQUESTED** [review]

File: client/src/components/RelatedAction.jsx [hunk 10]
```diff
@@ -1,21 +1,125 @@
...
 
-const CrossCircle = '../assets/outline_highlight_off_black_24dp.png'
+const CrossCircle= '../assets/outline_highlight_off_black_24dp.png'
 
-const StyleImg = styled.img`
+const StyleImg= styled.img`
 width: 50px;
 height: 50px;
 `
+const StyledModal= Modal.styled`
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: remove spaces next to =

File: client/src/components/RelatedAction.jsx [hunk 11]
```diff
@@ -1,21 +1,125 @@
...
+`
+
+
+export default function RelatedAction ({product, overviewProduct}) {
+  const [isRelated, setIsRelated] = useState(true);
+  const [isOpen, setIsOpen] = useState(false);
+  const [opacity, setOpacity] = useState(0);
+  const [compareInfo, setCompareInfo] = useState([1,2,3]);
+
+  const clickHandle= (e) => {
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: Remove unnecessary whitespace

File: client/src/components/RelatedAction.jsx [hunk 12]
```diff
@@ -1,21 +1,125 @@
...
+    toggleModal()
+    comparisionData()
+  }
+
+  const toggleModal= (e) => {
+    setOpacity(0);
+    setIsOpen(!isOpen);
+  }
+
+  const afterOpen= () => {
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: Add space after `afterOpen`

File: client/src/components/RelatedAction.jsx [hunk 13]
```diff
@@ -1,21 +1,125 @@
...
+    setTimeout(() => {
+      setOpacity(1);
+    }, 100);
+  }
 
-export default function RelatedAction () {
-  const [isRelated, setIsRelated] = useState(true)
+  const beforeClose= () => {
```
Comment (raphaelspies) [inline client/src/components/RelatedAction.jsx]: Add space after beforeClose

**Review by erikeh — APPROVED** [review]

**Comment by Lordnickim** [comment]
@raphaelspies Okay spacing should be good to go now lols

**Review by raphaelspies — APPROVED** [review]
